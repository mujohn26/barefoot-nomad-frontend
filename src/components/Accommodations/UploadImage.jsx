import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import CircularProgress from '@material-ui/core/CircularProgress';
import { log } from 'util';

const useStyles = makeStyles(theme => ({
	root: {
		'& > *': {
			margin: theme.spacing(1),
		},
	},
	input: {
		display: 'none',
	},
	button: {
		boxShadow: 'none',
		textTransform: 'none',
		fontSize: 16,
		width: '150px',
		height: '40px',
		lineHeight: 1.5,
		color: 'black',
		textAlign: 'center',
		['@media (max-width:750px)']: {
			width: '308.77px',
		},
	},
}));
export default function UploadImage(props) {
	const classes = useStyles();
	return (
		<div>
			<input
				data-test="upload-btn"
				accept='image/*'
				className={classes.input}
				id='contained_button_file'
				multiple
				type='file'
				onChange={e => {
					props.uploadAccommodationImage(e);
				}}
				display='hidden'
			/>
			<label
				style={{
					fontSize: 15,
					color: '#C4C4C4',
					textAlign: 'center',
				}}
			>
				Provide Accomodation Image
			</label>
			&ensp;
			<label htmlFor='contained_button_file'>
			{!props.isLoading?(
				<Button
					variant='contained'
				
					color='default'
					className={classes.button}
					component='span'
					startIcon={<CloudUploadIcon />}
				>
					Upload
				</Button>):(<CircularProgress className={classes.CircularProgress} />)}
			</label>
		</div>
	);
}
