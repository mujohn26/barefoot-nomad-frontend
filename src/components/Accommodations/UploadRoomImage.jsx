import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import { log } from 'util';

const useStyles = makeStyles(theme => ({
	root: {
		'& > *': {
			margin: theme.spacing(1),
		},
	},
	input: {
		display: 'none',
	},
	button: {
		boxShadow: 'none',
		textTransform: 'none',
		fontSize: 16,
		width: '150px',
		height: '40px',
		lineHeight: 1.5,
		color: 'black',
		textAlign: 'center',
		['@media (max-width:750px)']: {
			width: '308.77px',
		},
	},
}));
export function UploadRoomImage(props) {
	const classes = useStyles();
	return (
		<div>
			<input
				data-test='upload-btn'
				accept='image/*'
				className={classes.input}
				id={props.id}
				multiple
				type='file'
				onChange={e => {
					props.cardClicked(props.id);
					props.HandleUploadRoomImage(e, props.id);
				}}
				display='hidden'
			/>
			<label
				style={{
					fontSize: 15,
					color: '#C4C4C4',
					textAlign: 'center',
				}}
			>
				{props.label}
			</label>
			&ensp;
			<label htmlFor={props.id}>
				<Button
					variant='contained'
					color='default'
					className={classes.button}
					component='span'
					startIcon={<CloudUploadIcon />}
				>
					Upload
				</Button>
			</label>
		</div>
	);
}
