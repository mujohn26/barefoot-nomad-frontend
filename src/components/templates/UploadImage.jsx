import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles(theme => ({
	root: {
		'& > *': {
			margin: theme.spacing(1),
		},
	},
	input: {
		display: 'none',
	},
}));
export default function UploadImage(props) {
	const classes = useStyles();
	return (
		<div>
			<input
				data-test='upload-btn'
				accept='image/*'
				className={classes.input}
				id='contained_button_file'
				multiple
				type='file'
				onChange={e => {
					props.uploadImage(e);
				}}
				display='hidden'
			/>
			<label htmlFor='contained_button_file'>
				<Button
					variant='contained'
					color='primary'
					component='span'
					style={{
						backgroundColor: '#0094ff',
						borderRadius: 0,
					}}
				>
					Upload image
				</Button>
			</label>
		</div>
	);
}
