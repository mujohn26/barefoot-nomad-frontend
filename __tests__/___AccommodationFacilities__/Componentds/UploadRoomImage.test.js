import React from 'react';
import { shallow,mount } from 'enzyme';
import UploadImage from '../../../src/components/Accommodations/UploadImage.jsx';
import {UploadRoomImage} from '../../../src/components/Accommodations/UploadRoomImage.jsx';

describe('upload images', () => {
	it('should upload accommodation image on cloud', () => {
		const wrapper = shallow(<UploadImage uploadAccommodationImage={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
    });
    it('should upload room image on cloud', () => {
		const wrapper2 = shallow(<UploadRoomImage HandleUploadRoomImage={jest.fn()} />);
		expect(wrapper2).toMatchSnapshot();
    });
    it('should upload room image on cloud', () => {
        const wrapper2 = mount(<UploadRoomImage HandleUploadRoomImage={jest.fn()}  cardClicked={jest.fn()} />);
        const upload_image = wrapper2.find('[data-test="upload-btn"]');
		upload_image
			.first()
			.props()
			.onChange({}, {});

    })
    it('should upload accommodation image on cloud', () => {
        const wrapper2 = mount(<UploadImage uploadAccommodationImage={jest.fn()}  />);
        const upload_image = wrapper2.find('[data-test="upload-btn"]');
		upload_image
			.first()
			.props()
			.onChange({}, {});

    })

	// });
});
