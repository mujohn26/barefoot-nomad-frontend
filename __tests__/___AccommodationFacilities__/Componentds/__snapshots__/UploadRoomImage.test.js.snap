// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upload images should upload accommodation image on cloud 1`] = `
<div>
  <input
    accept="image/*"
    className="makeStyles-input-2"
    data-test="upload-btn"
    display="hidden"
    id="contained_button_file"
    multiple={true}
    onChange={[Function]}
    type="file"
  />
  <label
    style={
      Object {
        "color": "#C4C4C4",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Provide Accomodation Image
  </label>
   
  <label
    htmlFor="contained_button_file"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-3"
      color="default"
      component="span"
      startIcon={<Memo(CloudUploadIcon) />}
      variant="contained"
    >
      Upload
    </WithStyles(ForwardRef(Button))>
  </label>
</div>
`;

exports[`upload images should upload room image on cloud 1`] = `
<div>
  <input
    accept="image/*"
    className="makeStyles-input-5"
    data-test="upload-btn"
    display="hidden"
    multiple={true}
    onChange={[Function]}
    type="file"
  />
  <label
    style={
      Object {
        "color": "#C4C4C4",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  />
   
  <label>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-button-6"
      color="default"
      component="span"
      startIcon={<Memo(CloudUploadIcon) />}
      variant="contained"
    >
      Upload
    </WithStyles(ForwardRef(Button))>
  </label>
</div>
`;
